flink-core/src/test/java/org/apache/flink/api/common/typeutils/TypeSerializerSnapshotMigrationTestBase.java

```diff
@@ -452,11 +490,12 @@ protected boolean matchesSafely(TypeSerializerSchemaCompatibility<T> testResultC
 				if (expectedCompatibilty.isCompatibleAsIs()) {
 					return testResultCompatibility.isCompatibleAsIs();
 				} else if (expectedCompatibilty.isIncompatible()) {
-					return testResultCompatibility.isCompatibleAfterMigration();
-				} else if (expectedCompatibilty.isIncompatible()) {
-					return testResultCompatibility.isIncompatible();
+					return testResultCompatibility.isCompatibleAfterMigration() ||
+							testResultCompatibility.isIncompatible();
```

Comment:
looks a bit odd.

If think it's supposed to be
```
...
} else if (expectedCompatibilty.isIncompatible()) {
    return testResultCompatibility.isIncompatible();
} else if (expectedCompatibilty.isCompatibleAfterMigration()) {
    return testResultCompatibility.isCompatibleAfterMigration();
}
...
```

Reply:
yes, I think I merged this version in the end 😅 (I mean the correct one)